packages/engine-formula/src/engine/dependency/formula-dependency.ts

```diff
@@ -707,28 +689,32 @@ export class FormulaDependencyGenerator extends Disposable {
         }
     }
 
-    private _convertDirtyRangesToMap(dirtyRanges: IFeatureDirtyRangeType) {
-        const map = new Map<string, Map<string, IRange[]>>();
+    private _convertDirtyRangesToUnitRange(dirtyRanges: IFeatureDirtyRangeType) {
+        const unitRange: IUnitRange[] = [];
         for (const unitId in dirtyRanges) {
             const unitMap = dirtyRanges[unitId];
-            const unitRangeMap = new Map<string, IRange[]>();
             for (const subUnitId in unitMap) {
                 const ranges = unitMap[subUnitId];
-                unitRangeMap.set(subUnitId, ranges);
+                ranges.forEach((range) => {
```

Comment:
please use for loop or while loop insread array function forEach to ignore build function scope